File: training/rust/src/application/creature_util.rs
use crate::domain::model::fightable::Fightable;
use crate::domain::model::*;

pub struct CreatureUtil;

impl CreatureUtil {
    pub const STANDARD_ARMOR_CLASS_DESCRIPTION: &'static str = "Natural Armor";

    /* Counts the treasure hoard a creature guards. */
    pub fn treasure_hoard(creature: &Creature) -> u32 {
        creature.xp_value() * 10
    }

    pub fn fastest_speed(creature: &Creature) -> Option<Speed> {
        creature.speeds().values().copied().max_by_key(|speed| speed.speed())
    }
}

File: sample-projects/vue/src/de/sots/cellarsandcentaurs/application/CreatureUtil.ts
import * as model from '../domain/model/Creature'
import { Fightable } from '../domain/model/Fightable'

export class CreatureUtil {
  static readonly STANDARD_ARMOR_CLASS_DESCRIPTION: string = 'Natural Armor'

  /*
   * Counts the treasure hoard a creature guards.
   */
  static countHoard(creature: model.Creature): number {
    return (creature.getHitPoints()?.max ?? 0) * 10
  }
}
